src/blockdata/transaction.rs

```diff
@@ -220,12 +220,73 @@ impl Default for TxIn {
         TxIn {
             previous_output: OutPoint::default(),
             script_sig: Script::new(),
-            sequence: u32::max_value(),
+            sequence: Sequence::default(),
             witness: Witness::default(),
         }
     }
 }
 
+/// Bitcoin transaction input sequencing number.
+/// 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Sequence(pub u32);
```

Comment:
If this type is to abstract over nSequence do we want the inner int to be private, wait for others to comment though, don't go just off my review.

Reply:
I also thought the value doesn't need to be public but made it public following https://github.com/rust-bitcoin/rust-bitcoin/issues/1082#issuecomment-1172471235.

However, I do see the benefits of having the value public where users might want to do arithmetic with the value for any reason.

Reply:
Sequence has no invariants so I think public is fine.